cmd/oceantv/broadcast_states.go

```diff
@@ -571,7 +571,16 @@ func broadcastCfgToState(ctx *broadcastContext) state {
 }
 
 func createBroadcastAndRequestHardware(ctx *broadcastContext, cfg *BroadcastConfig, onCreation func() error) {
-	err := ctx.man.CreateBroadcast(
+	background := context.Background()
```

Comment:
This works OK, but I'm thinking it might be better if we put this in the hardware state machine for better encapsulation. We can check this in response to the handling of a hardwareStartRequestEvent, and if not OK, then we can emit a new event controllerFailedEvent or something along those lines. Then the broadcast state machine can handle this, send an email notification, and disable the broadcast. 

Also, using onFailureClosure as you might know performs the increment of the failure counter and disables after a few failures. I'm thinking on this failure we instead disable straight away. We can factor out the disabling code from the onFailureClosure so both mechanisms can use it.

Reply:
Thanks for the feedback, I thought I would just try something and get your thoughts on where it should go, as I have only a passing understanding of the full structure.

Reply:
No worries, you managed to do this quickly! It's good, but I think it could be slightly better :)

Reply:
Also, do you have any thoughts on broadcasts without controllers? Should we just skip the check if the controller MAC is empty?

Reply:
Yeah let's just skip the check in that case.

Reply:
Why are we not using the context that is supplied to the function, and instead using a background context?

Reply:
The context supplied to the function is a broadcast context, which I don't think implements the context interface (not that I think there is one). Could be wrong though.

 @saxon-milton?